apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/CustomElementMatchers.java

```diff
@@ -131,19 +131,34 @@ private static boolean canLoadClass(@Nullable ClassLoader target, String classNa
      */
     public static ElementMatcher.Junction<ProtectionDomain> implementationVersionLte(final String version) {
         return new ElementMatcher.Junction.AbstractBase<ProtectionDomain>() {
+            /**
+             * Returns true if the implementation version read from the manifest file referenced by the given
+             * {@link ProtectionDomain} is lower than or equal to the version set to this matcher
+             *
+             * @param protectionDomain a {@link ProtectionDomain} from which to look for the manifest file
+             * @return true if version parsed from the manifest file is lower than or equals to the matcher's version
+             * NOTE: returns true if matching fails, logging a warning message
+             */
             @Override
             public boolean matches(@Nullable ProtectionDomain protectionDomain) {
-                Version pdVersion = readImplementationVersionFromManifest(protectionDomain);
-                Version limitVersion = new Version(version);
-                if (pdVersion != null) {
-                    return pdVersion.compareTo(limitVersion) <= 0;
+                try {
+                    Version pdVersion = readImplementationVersionFromManifest(protectionDomain);
+                    Version limitVersion = new Version(version);
+                    if (pdVersion != null) {
+                        return pdVersion.compareTo(limitVersion) <= 0;
+                    }
+                } catch (Exception e) {
+                    logger.info("Cannot read implementation version based on ProtectionDomain. This should not affect " +
+                        "you agent's functionality. Failed with message: " + e.getMessage());
+                    logger.debug("Implementation version parsing error: " + protectionDomain, e);
                 }
                 return true;
             }
         };
     }
 
-    private static @Nullable Version readImplementationVersionFromManifest(@Nullable ProtectionDomain protectionDomain) {
+    private static @Nullable
```

Comment:
Annotate the method instead of the return type so that IntelliJ's auto-formatting does not split into two lines